src/ucp/api/ucp.h

```diff
@@ -1609,6 +1609,55 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
                            ucp_ep_h *ep_p);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Add user defined callback for active message.
+ *
+ * This routine adds a user defined callback to be used for ucp_am_put_nb.
+ *
+ * @param [in]  worker      UCP worker on which to set the am handler
+ * @param [in]  id          Active message id. Must be 0..(UCP_AM_ID_MAX - 1)
```

Comment:
maybe avoid UCP_AM_ID_MAX altogether?
how about defining ucp_am_id_t as uint16_t?

Reply:
Is there a reason for uint16_t? Like UCT, we keep an array of callbacks. However it attached to the ucp_worker opposed to the uct_iface. Having an array of 2^16 - 1 callbacks seems likes a lot.